akka-http/src/main/scala/akka/http/scaladsl/server/Route.scala

```diff
@@ -65,9 +73,31 @@ object Route {
                                 exceptionHandler: ExceptionHandler         = null): Flow[HttpRequest, HttpResponse, NotUsed] =
     Flow[HttpRequest].mapAsync(1)(asyncHandler(route))
 
+  def asyncHandler(route: Route, routingSettings: RoutingSettings, parserSettings: ParserSettings)(implicit system: ActorSystem): HttpRequest => Future[HttpResponse] = {
+    val routingLog = RoutingLog(system.log)
+    implicit val executionContext = system.dispatcher
+    implicit val materializer = Materializer.matFromSystem(new ClassicActorSystemProvider {
+      override val classicSystem = system
+    })
+    // We can seal more efficiently here because we know we can have no inherited settings:
+    val sealedRoute = {
+      import directives.ExecutionDirectives._
+      (handleExceptions(ExceptionHandler.default(routingSettings)) & handleRejections(RejectionHandler.default))
+        .tapply(_ => route)
+    }
+    request => {
```

Comment:
If that part is exactly the same as below in the old impl, could you extract it?